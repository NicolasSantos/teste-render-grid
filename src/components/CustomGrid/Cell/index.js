import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronUp, faChevronDown } from '@fortawesome/free-solid-svg-icons';
import { ORDER_TYPE } from '../../../types/sort';
import './index.scss';

const Cell = ({value, style, isFirstRow, onClickSort, sortColumn}) => {
    return (
        <div className={`cell ${isFirstRow && "cell--isFirstRow"}`} style={style}>
            <span className={"cell__value"} title={value}>{value}</span>
            {isFirstRow && 
                <div className="cell__sort" onClick={onClickSort ? () => onClickSort(value) : null}>
                    {   !sortColumn.columnName || sortColumn.columnName !== value || (sortColumn.columnName === value && sortColumn.order === ORDER_TYPE.DESC)
                        ? <FontAwesomeIcon icon={faChevronUp} size="xs"/> : ''
                    }

                    {   !sortColumn.columnName || sortColumn.columnName !== value || (sortColumn.columnName === value && sortColumn.order === ORDER_TYPE.ASC)
                        ? <FontAwesomeIcon icon={faChevronDown} size="xs"   /> : ''
                    }
                </div>
            }
        </div>
    )
}

export default Cell;