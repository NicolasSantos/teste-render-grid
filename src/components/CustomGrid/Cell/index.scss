@import '../../../styles/abstracts/variables.scss';

.cell {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    padding-right: 5px;

    &--isFirstRow {
        display: flex;
        font-weight: bold;

        .cell__value {
            width: 98%;
            border-bottom: 1px solid $cell-border;
            height: 80%;
            text-transform: capitalize;
        }

        .cell__sort {
            display: flex;
            flex-direction: column;
            position: absolute;
            right: 10px;
            top: -6px;
            height: 100%;
            justify-content: center;
            cursor: pointer;
        }
    }
}