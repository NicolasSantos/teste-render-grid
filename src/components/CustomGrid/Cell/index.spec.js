import React from "react";
import { shallow } from "enzyme";
import Cell from "./index";

describe("[COMPONENT] - Cell", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Cell/>);
    });

    it('Snapshot Cell', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Cell with isFirstRow class', () => {
        expect(wrapper.hasClass('cell--isFirstRow')).toBeFalsy();

        wrapper.setProps({ isFirstRow: true });

        expect(wrapper.hasClass('cell--isFirstRow')).toBeTruthy();
    });

    it('Cell with click on sort', () => {
        const mockCallback = jest.fn();

        wrapper.setProps({ isFirstRow: true, onClickSort: mockCallback });
        wrapper.find('.cell__sort').simulate('click');
        
        expect(mockCallback.mock.calls.length).toEqual(1);
    });
})