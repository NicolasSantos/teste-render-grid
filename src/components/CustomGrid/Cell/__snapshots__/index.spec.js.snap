// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] - Cell Snapshot Cell 1`] = `
<div
  className="cell undefined"
>
  <span
    className="cell__value"
  />
</div>
`;
